src/v/cloud_storage/topic_path_utils.cc

```diff
@@ -24,8 +24,21 @@ const std::regex labeled_manifest_path_expr{
   R"REGEX(meta/([^/]+)/([^/]+)/[^/]+/\d+/topic_manifest\.bin)REGEX"};
 } // namespace
 
+ss::sstring labeled_topic_manifests_root() { return "meta"; }
+
+chunked_vector<ss::sstring> prefixed_topic_manifests_roots() {
+    constexpr static auto hex_chars = std::string_view{"0123456789abcdef"};
+    chunked_vector<ss::sstring> roots;
+    roots.reserve(hex_chars.size());
+    for (char c : hex_chars) {
+        roots.emplace_back(fmt::format("{}0000000", c));
+    }
+    return roots;
+}
```

Comment:
is this a static list of 16 strings? seems ok unless its in the hot path in which case perhaps it'd be an easy thing to optimize.

Reply:
Yeah, it could be static/constexpr. Opted for brevity mostly, this is roughly what's used in topic_recovery_service today. Agreed though we can optimize in the future